plugins/GSdx/GS.cpp

```diff
@@ -481,11 +481,21 @@ EXPORT_C_(int) GSopen2(void** dsp, uint32 flags)
 	bool toggle_state = !!(flags & 4);
 
 	GSRendererType renderer = s_renderer;
+	GSRendererType INI_Renderer = static_cast<GSRendererType>(theApp.GetConfigI("Renderer"));
+
 	// Fresh start up or config file changed
+#ifdef _WIN32
 	if (renderer == GSRendererType::Undefined)
 	{
-		renderer = static_cast<GSRendererType>(theApp.GetConfigI("Renderer"));
+		renderer = (INI_Renderer == GSRendererType::Undefined) ? GSUtil::GetBestRenderer() : INI_Renderer;
 	}
+#else
+	if (renderer == GSRendererType::Undefined)
+	{
+		renderer = INI_Renderer;
+	}
+#endif
+
```

Comment:
Perhaps this could simply be:
```
	if (renderer == GSRendererType::Undefined)
	{
		renderer = static_cast<GSRendererType>(theApp.GetConfigI("Renderer"));
#ifdef _WIN32
		if (renderer == GSRendererType::Undefined) // Well, maybe GSRendererType::Default is better
			renderer = GSUtil::GetBestRenderer();
#endif
	}
```